### DISCUSSION TITLE
How to show a bidirectional transition between two states in state machine as two separate transitions ??

### DISCUSSION THREAD

**Discussion by Shivmatli** [networkx/networkx#6842] [Q&A]
I am new to NetworkX library. And I am trying to generate the state machine graph using networkx library commands. 

I have one state machine for example below

```python
# Importing the packages
import networkx as nx
import matplotlib.pyplot as plt

# Create a directed graph (DiGraph) for the state machine
G = nx.DiGraph()

# Add states to the graph
G.add_node("State_A_Standby")
G.add_node("State_A")
G.add_node("State_B")
G.add_node("State_C")


# Add transitions between states
G.add_edge("State_A_Standby", "State_A", label="Start system")
G.add_edge("State_A", "State_B", label="Transition 1")
G.add_edge("State_B", "State_C", label="Transition 2")
G.add_edge("State_C", "State_B", label="Transition 3")
G.add_edge("State_A", "State_A_Standby", label="Stop System")
G.add_edge("State_B", "State_A_Standby", label="Stop System")
G.add_edge("State_C", "State_A_Standby", label="Stop System")

# Draw the state machine diagram
pos = nx.spring_layout(G, seed=42)
nx.draw(G, pos, with_labels=True, node_size=2000, node_color='skyblue', font_size=10, font_weight='bold')
edge_labels = nx.get_edge_attributes(G, 'label')
nx.draw_networkx_edge_labels(G, pos, edge_labels=edge_labels, font_size=8)

# Show the diagram
plt.show()
```

for example here there are transitions from State_B to State_C and from State_C to State_B. 

Here I want to show these transition separatly between states State_B and State_C. 

Moreover I want to generate the different possible paths as well between State_A_Standby to itself State_A_Standby.

So, anyone have any idea about how to proceed for these two tasks ??

**Comment by dschult** [comment]
I am not sure what you are referring to here. Both arrows are drawn and labeled for your code. But they apear on top of each other so you can't see it very well. I'm assuming that is the trouble.

You should try the separate drawing commands for more tight control: `draw_networkx_nodes`, `draw_networkx_edges` and `draw_networkx_edge_labels`. And you should try the edge option e.g. `connectionstyle='arc3,rad=0.2'`. And you will probably want something like `label_pos=0.3` to separate the edge labels. For even more control you can draw some edges with one function and other edges with other functions.

Take a look at the [gallery](https://networkx.org/documentation/stable/auto_examples/index.html) examples in the docs and the [drawing](https://networkx.org/documentation/stable/reference/drawing.html) section of the reference docs.

**Reply by Shivmatli** [reply]
I have tried like many ways but it's not working. Can you please give me a small example of the graph which has two transition back and forth between two nodes. In graph i want to see both the transitions separatly. If you can give me a small example of like 3 or 4 nodes and some transitions, then it would be a great help sir.

Thank you so much for your last response.

**Reply by dschult** [reply]
How about:
```python
import networkx as nx
import matplotlib.pyplot as plt

G = nx.DiGraph([(0, 1), (1, 0)])
pos=nx.spring_layout(G)
edge_labels = {(0,1): "(0,1)", (1,0): "(1,0)"}

plt.clf()
nx.draw_networkx_nodes(G, pos=pos)
nx.draw_networkx_labels(G, pos=pos)
nx.draw_networkx_edges(G, pos=pos, arrows=True, connectionstyle='arc3,rad=0.2')
nx.draw_networkx_edge_labels(G, pos=pos, edge_labels=edge_labels, label_pos=0.3)
plt.show()
```
